rest-service/manager_rest/deployment_update/manager.py

```diff
@@ -573,6 +608,60 @@ def _execute_workflow(self,
         )
         return new_execution
 
+    def _extract_plugins_changes(self, dep_update):
+        """Extracts plugins that need to be installed and uninstalled.
+
+        :param dep_update: a DeploymentUpdate object.
+        :return: plugins that need installation and uninstallation (a tuple).
+        """
+
+        def is_centrally_deployed(plugin):
+            return plugin[constants.PLUGIN_EXECUTOR_KEY] \
+                   == constants.CENTRAL_DEPLOYMENT_AGENT
+
+        def append_plugins_to_list(source_dict, key_list, target_list):
+            for k in key_list:
+                target_list.append(source_dict[k])
+
+        deployment = self.sm.get(models.Deployment, dep_update.deployment_id)
+        old_plan = deployment.blueprint.plan
+        new_plan = dep_update.deployment_plan
+        # Plugins to install in the old plan
+        plugins_to_install_old = extract_and_merge_plugins_to_install(
+            old_plan[constants.DEPLOYMENT_PLUGINS_TO_INSTALL],
+            old_plan[constants.WORKFLOW_PLUGINS_TO_INSTALL],
+            filter_func=is_centrally_deployed)
+        # Plugins to install in the new plan
+        plugins_to_install_new = extract_and_merge_plugins_to_uninstall(
```

Comment:
same code for both cases, so why not add this to 'extract_and_merge_plugins_to_uninstall'

Reply:
I think you didn't consider that the `extract_and_merge` functions are using [here](https://github.com/cloudify-cosmo/cloudify-manager/blob/CY-1060-plugin-update/workflows/cloudify_system_workflows/deployment_environment.py#L40) and [here](https://github.com/cloudify-cosmo/cloudify-manager/blob/CY-1060-plugin-update/workflows/cloudify_system_workflows/deployment_environment.py#L81)

Reply:
You don't have to put that there, I meant that there is a repetition that u can encapsulate